### PR TITLE
Changed configuration to support multiple repositories and topic

### PR DESCRIPTION
**Pull request by giampaolotrapasso**
Changed configuration to support multiple repositories and topic (issue #2).
I did not create a single client. First idea was to create a static singleton, but not sure how it would work on a distributed connector. I need to check better how connector works on a distributed environment.

In case, it can be done one a different issue.
Let me know what do you think.

### CODE REVIEW

File: src/main/java/com/simplesteph/kafka/GitHubSourceConnector.java [hunk 1]
```diff
@@ -30,10 +30,25 @@ public void start(Map<String, String> map) {
...
 
     @Override
-    public List<Map<String, String>> taskConfigs(int i) {
+    public List<Map<String, String>> taskConfigs(int n) {
+        List<GithubSourceTaskConfig> taskConfigs = config.getTasksConfig();
+
+        Integer maxRepositories = Math.min(n, taskConfigs.size());
+
+        if (maxRepositories < taskConfigs.size())
+            log.warn("Less workers than repositories, only first %d repositories will be extracted", maxRepositories);
```
Comment (simplesteph) [inline src/main/java/com/simplesteph/kafka/GitHubSourceConnector.java:39]: one task should definitely be able to extract multiple repo. Just create as many tasks as repos and let the framework take care of distributing that to the connect workers.

**Review by simplesteph — CHANGES_REQUESTED** [review]
overall looks good except on the creation of tasks. I believe you can create as many task configs as you want and then the connector will happily work even if one worker (to be tested)
Other thing I'm worried about is backwards compatibility

**Comment by giampaolotrapasso** [comment]
Ok, now I see your point. Instead of splitting configuration in multiple tasks, you want to keep the code as before but allowing task to query for multiple repositories. 

So I have a couple of questions before changing the PR:

**first**. backward compatibility is important, so in case of a configuration like this
```
topic=github-issues
github.owner=kubernetes
github.repo=kubernetes
github.repositories=scala/scala:lang-topic,apache/kafka:kafka-topic,apache/cassandra:cassandra-topic
```
there are 3 possible choices:
1. ignoring old configuration
2. ignoring new configuration (backward compatibility)
3. merging the repos (backward compatibility)
are you for 2 or 3?

then **second** questions is: 
If task is going to read from multiple repos, which kind of policy should be adopted? Round robin? Being async on futures? What would you suggest?

Let me know and thank you meanwhile for the review and your time.

**Comment by simplesteph** [comment]
1. backwards compatibility: new configs should precede over old config (if both are supplied), we deprecate the old config, but we keep both for now

2. I looked back at my comment and it may not make sense. I like your approach in some ways. I think there are two options. 
a. ignore the maxTasks parameters. I sort of like this, if we outline it in the documentation. Simplifies user config too
b. do your approach, ignoring some repo (may lead to unexpected behaviour)
c. do what JdbcConnector does: https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/JdbcSourceConnector.java#L151 
and each task can pull from multiple repos (round robin as you said, or multiple futures and awaiting on them).

To be honest I'm leaning towards `2.a.`
What do you think?

**Pull request closed without merging**